### PR TITLE
chore: Remove String Concat With `+` Operator

### PR DESCRIPTION
**Pull request by notJoon**
# Description

* modified the part where strings were concatenated using the `+` operator to use a formatting function instead.
    - This change was made for readability rather than performance concerns (generally, using the `+` operator for string concatenation can result in minor performance degradation). 

* extracted the logic for splitting strings with a specific separator into a common utility.

### CODE REVIEW

File: router/comptue_routes.gno [hunk 1]
```diff
@@ -180,7 +181,7 @@ func findCandidatePools() []PoolWithMeta {
 			token0Path,
 			token1Path,
 			pFee,
-			token0Path + ":" + token1Path,
+			ufmt.Sprintf("%s:%s", token0Path, token1Path),
```
Comment (r3v4s) [inline router/comptue_routes.gno:184]: why not use string builder here?

File: gov/governance/_GET_vote.gno [hunk 2]
```diff
@@ -110,12 +110,9 @@ func GetVotedAtByVoteKey(voteKey string) uint64 {
 }
 
 func divideVoteKeyToProposalIdAndUser(voteKey string) (uint64, std.Address) {
-	parts := strings.Split(voteKey, ":")
-	if len(parts) != 2 {
-		panic(addDetailToError(
-			errInvalidInput,
-			ufmt.Sprintf("_GET_vote.gno__divideVoteKeyToProposalIdAndUser() || voteKey(%s) is invalid", voteKey),
```
Comment (r3v4s) [inline gov/governance/_GET_vote.gno:117]: I really don't think removing detail error message is good idea.
It makes us hard to why it failed.
@onlyhyde Any thoughts?

**Review by r3v4s — CHANGES_REQUESTED** [review]
IMHO, removing detail error message should not be happened.
@onlyhyde WDYT

File: gov/governance/_GET_vote.gno (hunk 2 above)
Comment (onlyhyde) [reply gov/governance/_GET_vote.gno:117]: yes i agree with @r3v4s . error message is already confirmed our policy. so, we have to use previous error message

File: router/comptue_routes.gno (hunk 1 above)
Comment (notJoon) [reply router/comptue_routes.gno:184]: in this case, these is no significant difference between using strings.Builder and formatting function. so I use latter one.

File: gov/governance/_GET_vote.gno (hunk 2 above)
Comment (notJoon) [reply gov/governance/_GET_vote.gno:117]: reverted for panic messages

https://github.com/gnoswap-labs/gnoswap/pull/394/commits/1b17b6c4312417f01c59f084130cce786b482c45

**Review by r3v4s — APPROVED** [review]

**Review by onlyhyde — APPROVED** [review]

**Pull request merged**